invoices/invoice_expiry_watcher.go

```diff
@@ -151,18 +240,67 @@ func makeTimestampExpiry(paymentHash lntypes.Hash,
 	}
 }
 
+// makeHeightExpiry creates height-based expiry entries using the lowest active
+// htlc's height.
+func makeHeightExpiry(paymentHash lntypes.Hash,
+	expiryHeights []uint32) *invoiceExpiryHeight {
+
+	var minHeight uint32
+
+	for _, height := range expiryHeights {
+		if height < minHeight || minHeight == 0 {
+			minHeight = height
+		}
+	}
+
+	// If we don't have any htlcs that are going to expire, we
+	// don't need to add this invoice to the expiry watcher yet.
+	if minHeight == 0 {
```

Comment:
I think we have serious bugs elsewhere if a htlc has a 0 min-height. Added a warning log in this case, and moved to pass in minimum height rather than `[]uint32`.